psutil/_psutil_osx.c

```diff
@@ -1798,6 +1799,57 @@ psutil_cpu_stats(PyObject *self, PyObject *args) {
 }
 
 
+/*
+ * Return temperatures of hardware components.
+ */
+static PyObject *
+psutil_sensors_temperatures(PyObject *self, PyObject *args) {
+    PyObject *py_retdict = PyDict_New();
+    // CPU only stats
+    PyObject *py_cpu_temp = PyDict_New();
+    PyObject *py_cpu_temp_current = NULL;
+    PyObject *py_cpu_temp_high = NULL;
+
+    if (py_retdict == NULL)
+        goto error;
+
+    if (py_cpu_temp == NULL)
+        goto error;
+
+    py_cpu_temp_current = Py_BuildValue("d", SMCGetTemperature(SMC_KEY_CPU_TEMP));
+    if (!py_cpu_temp_current)
+        goto error;
+
+    if (PyDict_SetItemString(py_cpu_temp, "current", py_cpu_temp_current)) {
+        goto error;
+    }
+
+    py_cpu_temp_high = Py_BuildValue("d", SMCGetTemperature(SMC_KEY_CPU_TEMP_HIGH));
+    if (!py_cpu_temp_high)
+        goto error;
+
+    if (PyDict_SetItemString(py_cpu_temp, "high", py_cpu_temp_high)) {
+        goto error;
+    }
+    if (PyDict_SetItemString(py_retdict, "TC0F", py_cpu_temp)) {
```

Comment:
what's "TC0F"?

Reply:
TC0F is the best SMC key that I have found to correlate to the CPU die temperature (and is the key read in smc.c), but again, there is not official documentation for this. This can be change to `CPU die` or left as is. More SMC keys could be added in the future.

Reply:
What's CPU die temperature? Also, what if the system has more than one CPU?

Reply:
This [https://app.assembla.com/spaces/fakesmc/wiki/Known_SMC_Keys]() is the best source I have found the documents this. There `CPU die` is marked as `TC0D`, but this does not exist on both systems I have tested. `TC0F` is best correlated to the temperature reported by the `Intel power gadget`

Reply:
Applying same logic as per-core temperature,  in multi-CPU systems there should be an equivalent `TC\dD`  key for per CPU, I'll see if I can make a loop over all that can be detected. But I do not have access to a multi-CPU system. (Also I am not sure that such a system even exists, and whether Apple themselves addressed this issue)